app/src/main/kotlin/com/github/gotify/NotificationSupport.kt

```diff
@@ -1,56 +1,167 @@
 package com.github.gotify
 
 import android.app.NotificationChannel
+import android.app.NotificationChannelGroup
 import android.app.NotificationManager
+import android.content.Context
 import android.graphics.Color
 import android.os.Build
 import androidx.annotation.RequiresApi
+import androidx.preference.PreferenceManager
+import com.github.gotify.client.model.Application
 import com.github.gotify.log.Log
 
 internal object NotificationSupport {
     @RequiresApi(Build.VERSION_CODES.O)
-    fun createChannels(notificationManager: NotificationManager) {
-        try {
-            // Low importance so that persistent notification can be sorted towards bottom of
-            // notification shade. Also prevents vibrations caused by persistent notification
-            val foreground = NotificationChannel(
-                Channel.FOREGROUND,
-                "Gotify foreground notification",
-                NotificationManager.IMPORTANCE_LOW
-            )
-            foreground.setShowBadge(false)
+    fun createForegroundChannel(context: Context, notificationManager: NotificationManager) {
+        // Low importance so that persistent notification can be sorted towards bottom of
+        // notification shade. Also prevents vibrations caused by persistent notification
+        val foreground = NotificationChannel(
+            Channel.FOREGROUND,
+            context.getString(R.string.notification_channel_title_foreground),
+            NotificationManager.IMPORTANCE_LOW
+        ).apply {
+            setShowBadge(false)
+        }
+        notificationManager.createNotificationChannel(foreground)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun createChannels(
+        context: Context,
+        notificationManager: NotificationManager,
+        applications: List<Application>
+    ) {
+        if (areAppChannelsRequested(context)) {
+            notificationManager.notificationChannels.forEach { channel ->
+                if (channel.id != Channel.FOREGROUND) {
+                    notificationManager.deleteNotificationChannel(channel.id)
+                }
+            }
```

Comment:
We probably don't want to delete existing channels that will continue to exist, because it will reset the configured customizations from the user.

Reply:
The thought was that if you delete an application you don't want the notification channels continue to exist.  
Therefore we need to delete them

Reply:
> It's the same story as with updating channels I think. The channel will just be hidden and can be restored at a later time preserving all user changes.

Exactly. The channels will never be deleted, notice the 'x categories deleted' at the bottom of the notification settings.  
BUT the customizations are bound to the application id.  
I suppose that if you delete an application and create a new one for other purposes the system restores settings for the old application.  
But I don't see a real problem there.

Reply:
Tbh I don't know about that.  
If that's the case, the problem's gone.

Reply:
Okay, I didn't know, seems counterintuitive that android saves all notification channels, even the ones that were deleted on the app.